src/sage.ts

```diff
@@ -7,45 +7,56 @@ import { DB, BOT, PREFIX, GITHUB_TOKEN } from '@root/config';
 import { Octokit } from '@octokit/rest';
 import { version as sageVersion } from '@root/package.json';
 import { registerFont } from 'canvas';
+import { SageData } from '@lib/types/SageData';
 
 consoleStamp(console, {
 	pattern: 'dd/mm/yy hh:MM:ss.L tt',
 	label: false
 });
 
-const bot = new Client({
-	fetchAllMembers: true,
-	disableMentions: 'everyone',
-	allowedMentions: { parse: ['users'] }
-});
+async function main() {
+	const bot = new Client({
+		fetchAllMembers: true,
+		disableMentions: 'everyone',
+		allowedMentions: { parse: ['users'] }
+	});
 
-MongoClient.connect(DB.CONNECTION, { useUnifiedTopology: true }).then((client) => {
-	bot.mongo = client.db(BOT.NAME);
-});
+	await MongoClient.connect(DB.CONNECTION, { useUnifiedTopology: true }).then((client) => {
+		bot.mongo = client.db(BOT.NAME);
+	});
 
-bot.login(BOT.TOKEN);
+	bot.login(BOT.TOKEN);
 
-bot.octokit = new Octokit({
-	auth: GITHUB_TOKEN,
-	userAgent: `Sage v${sageVersion}`
-});
-
-bot.once('ready', async () => {
-	const pieceFiles = readdirRecursive(`${__dirname}/pieces`);
-	for (const file of pieceFiles) {
-		const piece = await import(file);
-		const dirs = file.split('/');
-		const name = dirs[dirs.length - 1].split('.')[0];
-		if (typeof piece.default !== 'function') throw `Invalid piece: ${name}`;
-		piece.default(bot);
-		console.log(`${name} piece loaded.`);
-	}
+	bot.octokit = new Octokit({
+		auth: GITHUB_TOKEN,
+		userAgent: `Sage v${sageVersion}`
+	});
 
 	registerFont(`${__dirname}/../../assets/Roboto-Regular.ttf`, { family: 'Roboto' });
 
-	console.log(`${BOT.NAME} online`);
-	console.log(`${bot.ws.ping}ms WS ping`);
-	console.log(`Logged into ${bot.guilds.cache.size} guilds`);
-	console.log(`Serving ${bot.users.cache.size} users`);
-	bot.user.setActivity(`${PREFIX}help`, { type: 'PLAYING' });
-});
+	bot.once('ready', async () => {
+		const pieceFiles = readdirRecursive(`${__dirname}/pieces`);
+		for (const file of pieceFiles) {
+			const piece = await import(file);
+			const dirs = file.split('/');
+			const name = dirs[dirs.length - 1].split('.')[0];
+			if (typeof piece.default !== 'function') throw `Invalid piece: ${name}`;
+			piece.default(bot);
+			console.log(`${name} piece loaded.`);
+		}
+
+		console.log(`${BOT.NAME} online`);
+		console.log(`${bot.ws.ping}ms WS ping`);
+		console.log(`Logged into ${bot.guilds.cache.size} guilds`);
+		console.log(`Serving ${bot.users.cache.size} users`);
+
+		const status = (await bot.mongo.collection(DB.CLIENT_DATA).findOne({ _id: bot.user.id }) as SageData)?.status;
+
+		const activity = status?.name || `${PREFIX}help`;
+		const type = status?.type || 'PLAYING';
+		bot.user.setActivity(`${activity} (v${sageVersion})`, { type });
+		setTimeout(() => bot.user.setActivity(activity, { type }), 30e3);
```

Comment:
Why not just have the status always be `Playing s;help vX.X.X`?? Honestly, I like that better. Up to you though

Reply:
My thoughts where since `s;help` is a valid command but `s;help (v x.x.x)` is not. Having an invalid command in the status all the time may confuse the three people who actually look at it.

Reply:
Gotcha, makes sense.